opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,475 @@
+ */
+typedef int (*opal_accelerator_base_module_memmove_fn_t)(
+    int dev_id, void *dest, const void *src, size_t size,
+    opal_accelerator_transfer_type_t type);
+
+/**
+ * Allocates size bytes memory from the device and sets ptr to the
+ * pointer of the allocated memory. The memory is not initialized.
+ *
+ * @param[IN] dev_id         The device to allocate memory on
+ * @param[OUT] ptr           Returns pointer to allocated memory
+ * @param[IN] size           Size of memory to allocate
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_malloc_fn_t)(
+    int dev_id, void **ptr, size_t size);
+
+/**
+ * Frees the memory space pointed to by ptr which has been returned by
+ * a previous call to an opal_accelerator_base_module_malloc_fn_t().
+ * If the function is called on a ptr that has already been freed,
+ * undefined behavior occurs. If ptr is NULL, no operation is performed,
+ * and the function returns OPAL_SUCCESS.
+ *
+ * @param[IN] ptr            Pointer to free
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_free_fn_t)(
+    void *ptr);
+
+/**
+ * Retrieves the base address and/or size of a memory allocation of the
+ * device.
+ *
+ * @param[IN] ptr            Pointer to device memory to get base/size from
+ * @param[OUT] base          Base address of the memory allocation
+ * @param[OUT] size          Size of the memory allocation
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_get_address_range_fn_t)(
+    const void *ptr, void **base, size_t *size);
+
+/**
+ * Queries whether the device supports Inter Process Communication
+ * or not. If true, the functions:
+ *
+ * opal_accelerator_base_module_get_handle_size_fn_t()
+ * opal_accelerator_base_module_get_handle_fn_t()
+ * opal_accelerator_base_module_open_handle_fn_t()
+ * opal_accelerator_base_module_close_handle_fn_t()
+ *
+ * must be implemented.
+ *
+ * @return true              IPC supported
+ * @return false             IPC not supported
+ */
+typedef bool (*opal_accelerator_base_module_is_ipc_enabled_fn_t)(void);
+
+/**
+ * Returns the size of the Inter Process Communication memory handle
+ *
+ * @return size_t            Size of the IPC memory handle
+ */
+typedef size_t (*opal_accelerator_base_module_get_handle_size_fn_t)(void);
+
+/**
+ * Gets an interprocess memory handle for an existing device memory allocation.
+ *
+ * @param[IN] dev_ptr        Base pointer to previously allocated device
+ *                           memory
+ * @param[OUT] handle        Pointer to user allocated mem handle to return
+ *                           the memory handle in.
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ *
+ */
+typedef int (*opal_accelerator_base_module_get_handle_fn_t)(
```

Comment:
I was confused by what handle it is by reading the libfabric's implementation. If it is for IPC only, can we rename it as `*get_ipc_handle`? If it's not for IPC only, what are the other user cases for this function?

Reply:
I can do that

Reply:
Moved IPC out of this PR.